version: '3.3'
services:       
    rabbitmq:
        ports: 
            - "4369:4369"
            - "5671:5671"
            - "5672:5672"
            - "25672:25672"
            - "15671:15671"
            - "15672:15672"
        healthcheck:
            test: ["CMD", "curl", "-f", "http://localhost:15672"]
            interval: 30s
            timeout: 10s
            retries: 5
        image: rabbitmq:latest
        networks:
            - custom-network
        
    redis: 
        ports: 
            - '6379:6379'
        image: redis:latest
        networks:
            - custom-network
            
    mysql: 
        ports: 
            - '3333:3306'
        volumes: 
            - mysql-conf:/etc/mysql/conf.d
            - mysql-storage:/var/lib/mysql
        restart: on-failure
        environment: 
            MYSQL_ROOT_PASSWORD: admin
        image: mysql:latest
        networks:
            - custom-network
        
    grafana:
        ports:
            - '3000:3000'
        restart: on-failure
        depends_on: 
            - mysql
        volumes:
            - grafana-storage:/var/lib/grafana
        image: grafana/grafana:latest
        user: "472"
        networks:
            - custom-network
            
networks: 
    custom-network:
        driver: bridge
        name: custom-network
        
volumes:
  grafana-storage:
  mysql-storage:
  mysql-conf: